### PR TITLE
[Feature] support read map column from parquet file

### LINKED ISSUE #10677: Support struct/map for Data Lake
This issue is to lay the groundwork for supporting complex types (struct/map) for StarRocks on data lakes.

The goals of this issue:
- support complex types (struct/map) for hive/hudi/iceberg
- support reading complex types (struct/map) from orc/parquet/csv
- support map functions
- support pushdown subfields of struct

The subtasks for this issue:
- [ ] support reading full map column from hive/orc https://github.com/StarRocks/starrocks/pull/10546
- [ ] support reading full struct column from hive/orc
- [ ] support `[]` operator for map type
- [ ] support `.` operator for struct type
- [ ] support pushdown subfields of struct type
- [ ] support map functions
- [ ] support reading map/struct column from parquet/csv
- [ ] support reading map/struct column from iceberg/hudi

### PR DESCRIPTION
**Pull request by zombee0**
support read map column from parquet file

Signed-off-by: zombee0 <flylucas_10@163.com>

## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #10677 

## Problem Summary(Required) ：

there are two main change:
1. generate the IORange for map type column, the map type in parquet is represented by MAP<Struct<Key,Value>>, so we should get IORange by 
```
for (auto& field : schema_node->children[0].children) {
        int64_t range_end = 0;
        _collect_field_io_range(field, ranges, &range_end);
        end = std::max(end, range_end);
  }
```
2. The MapColumnReader, It is very similar to ListColumnReader, Just keep Map<Struct<key,value>> in mind.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Review by dirtysalt — DISMISSED** [review]

File: be/src/formats/parquet/column_reader.cpp [hunk 1]
```diff
@@ -178,6 +255,17 @@ Status ColumnReader::create(const ColumnReaderOptions& opts, const ParquetField*
         std::unique_ptr<ListColumnReader> reader(new ListColumnReader(opts));
         RETURN_IF_ERROR(reader->init(field, std::move(child_reader)));
         *output = std::move(reader);
+    } else if (field->type.type == TYPE_MAP) {
+        std::unique_ptr<ColumnReader> key_reader;
+        // ParquetFiled Map -> Map<Struct<key,value>>
+        RETURN_IF_ERROR(
+                ColumnReader::create(opts, &(field->children[0].children[0]), col_type.children[0], &key_reader));
```
Comment (DorianZheng) [inline be/src/formats/parquet/column_reader.cpp:262]: do we need some `DCHECK` for verifying the type of children and the num of children, etc.

File: be/src/formats/parquet/column_reader.cpp [hunk 2]
```diff
@@ -167,9 +168,85 @@ class ListColumnReader : public ColumnReader {
...
+
+    Status init(const ParquetField* field, std::unique_ptr<ColumnReader> key_reader,
+                std::unique_ptr<ColumnReader> value_reader) {
+        _field = field;
+        _key_reader = std::move(key_reader);
+        _value_reader = std::move(value_reader);
+        return Status::OK();
+    }
+
+    Status prepare_batch(size_t* num_records, ColumnContentType content_type, vectorized::Column* dst) override {
```
Comment (DorianZheng) [inline be/src/formats/parquet/column_reader.cpp:184]: plz add some comments

File: be/src/formats/parquet/column_reader.cpp [hunk 3]
```diff
@@ -167,9 +168,85 @@ class ListColumnReader : public ColumnReader {
...
+            map_column = down_cast<vectorized::MapColumn*>(dst);
+        }
+        auto* key_column = map_column->keys_column().get();
+        auto st = _key_reader->prepare_batch(num_records, content_type, key_column);
+        if (!st.ok() && !st.is_end_of_file()) {
+            return st;
+        }
+
+        auto* value_column = map_column->values_column().get();
+        st = _value_reader->prepare_batch(num_records, content_type, value_column);
```
Comment (DorianZheng) [inline be/src/formats/parquet/column_reader.cpp:203]: do we need to check if num of `key_column` equals to num of `value_column`

**Review by DorianZheng — DISMISSED** [review]
left comments, rest LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/src/formats/parquet/column_reader.cpp (hunk 3 above)
Comment (zombee0) [reply be/src/formats/parquet/column_reader.cpp:203]: I think parquet will check and promise it. I will add DCHECK used for debug

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by dirtysalt — APPROVED** [review]

**Review by DorianZheng — APPROVED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Review by Youngwb — APPROVED** [review]

**Pull request merged**